src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -102,24 +135,42 @@ export class CellExecutionQueue {
     }
     private async executeQueuedCells() {
         const notebook = await this.notebookPromise;
-        this.queueOfCellsToExecute.forEach((exec) => traceCellMessage(exec.cell, 'Ready to execute'));
-        while (this.queueOfCellsToExecute.length) {
+        this.queueToExecute.forEach((exec) => {
+            if (exec.cellExecution) {
+                traceCellMessage(exec.cellExecution.cell, 'Ready to execute');
+            } else if (exec.codeExecution) {
+                traceInfo('Ready to execute hidden code', exec.codeExecution.code.substring(0, 50));
+            }
+        });
+        while (this.queueToExecute.length) {
             // Take the first item from the queue, this way we maintain order of cell executions.
             // Remove from the queue only after we process it
             // This way we don't accidentally end up queueing the same cell again (we know its in the queue).
-            const cellToExecute = this.queueOfCellsToExecute[0];
-            traceCellMessage(cellToExecute.cell, 'Before Execute individual cell');
+            const toExecute = this.queueToExecute[0];
+            if (toExecute.cellExecution) {
+                traceCellMessage(toExecute.cellExecution.cell, 'Before Execute individual cell');
+            } else if (toExecute.codeExecution) {
+                traceInfo('Before Execute hidden code', toExecute.codeExecution.code.substring(0, 50));
+            }
```

Comment:
We have missed an else clause and that results in bugs, the suggestion I made will ensure typescript types will prevent such bugs